api/proxy/metrics/server.go

```diff
@@ -0,0 +1,27 @@
+package metrics
+
+import (
+	"net"
+	"strconv"
+
+	ophttp "github.com/ethereum-optimism/optimism/op-service/httputil"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+// Config ... Metrics server configuration
+type Config struct {
+	Host    string
+	Port    int
+	Enabled bool
+}
+
+func NewServer(registry *prometheus.Registry, cfg Config) *ophttp.HTTPServer {
+	address := net.JoinHostPort(cfg.Host, strconv.Itoa(cfg.Port))
+
+	h := promhttp.InstrumentMetricHandler(
+		registry, promhttp.HandlerFor(registry, promhttp.HandlerOpts{}),
+	)
+
+	return ophttp.NewHTTPServer(address, h)
```

Comment:
what does this ophttp provide? If its small can we copy the code into our repo? Otherwise can we just switch to using the golang http library directly

Reply:
https://github.com/ethereum-optimism/optimism/blob/develop/op-service/httputil/server.go

It wraps http.Server and adds some additional connivance functions. Plan to remove in a future PR.